demos/imagenet_classification_demo/classification_results.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <utility>
+
+#include <ie_blob.h>
+
+class ClassificationResult {
+public:
+    explicit ClassificationResult(InferenceEngine::Blob::Ptr output_blob,
+                                  size_t batch_size = 1,
+                                  size_t num_of_top = 10,
+                                  std::vector<std::string> image_names = {},
+                                  std::vector<std::string> labels = {}) :
+            _nTop(num_of_top),
+            _outBlob(std::move(output_blob)),
+            _labels(std::move(labels)),
+            _imageNames(std::move(image_names)),
+            _batchSize(batch_size) {
+    }
+
+    std::vector<unsigned> topResults(unsigned int n, InferenceEngine::Blob& input) {
+        std::vector<unsigned> output;
+
+        switch (input.getTensorDesc().getPrecision()) {
+            case InferenceEngine::Precision::FP32: {
+                using currentBlobType = InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP32>::value_type;
+                InferenceEngine::TBlob<currentBlobType>& tblob = dynamic_cast<
+                                                                    InferenceEngine::TBlob<currentBlobType>&>(input);
+                topResults(n, tblob, output);
```

Comment:
The wrapping function (`std::vector<unsigned> topResults(unsigned int n, InferenceEngine::Blob& input)`) is not needed.
Blob shouldn't be expected to be anything else but FP32. Usually CPUs can't even process FP16, so `void topResults(unsigned int n, InferenceEngine::TBlob<T> &input, std::vector<unsigned> &output)` shouldn't be templated.

After that there will be only one function in the file. I suggest moving that function into `main.cpp`.